src/api/c/memoryapi.hpp

```diff
@@ -0,0 +1,23 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <af/event.h>
+#include <af/memory.h>
+
+typedef struct {
+  void *ptr;
+  af_event event;
+} af_memory_event_pair_t;
```

Comment:
`af_memory_event_pair_t` is internal, better to use CamelCase to be consistent with other internal objects. This will also remove confusion caused by `_t` suffix - it is easy to miss that.

Perhaps `MemoryEventPair` ?

Actually, can't we just use `using MemoryEventPair = std::pair<void*, af_event>;` instead ? If it is going to be just two elements, used internally and no member functions.

Reply:
As I mentioned, this needs to be used externally, since it's part of the public API for the memory manager, since `MemoryManager::alloc` will need to return one.

Reply:
I misread your comment - we can get rename `af_memory_event_pair_t` and typedef to be a `std::pair`. This makes sense.